lib/elixir_analyzer/test_suite/dna_encoding.ex

```diff
@@ -0,0 +1,26 @@
+defmodule ElixirAnalyzer.TestSuite.DNA do
+  @moduledoc """
+  This is an exercise analyzer extension module for the DNA Encoding concept exercise
+  """
+
+  use ElixirAnalyzer.ExerciseTest
+  alias ElixirAnalyzer.Constants
+
+  assert_no_call "does not call Enum.reduce" do
+    type :essential
+    called_fn module: Enum, name: :reduce
+    comment Constants.dna_encoding_use_recursion()
+  end
+
+  assert_no_call "does not call List.foldr" do
+    type :essential
+    called_fn module: List, name: :foldr
+    comment Constants.dna_encoding_use_recursion()
+  end
+
+  assert_no_call "does not call the for/1 function" do
+    type :essential
+    called_fn name: :for
```

Comment:
Here, students would still be able to use it in the form of `Kernel.SpecialForms.for`. I know it's a bit of  stretch that they would do that, but there is a simple way to add it. First, change to

```suggestion
    called_fn module: Kernel.SpecialForm, name: :for
```

Then we have to make sure that the `Kernel.SpecialForm` functions are imported, you can do that here:
https://github.com/exercism/elixir-analyzer/blob/71130beddcab0da34896ea41f24ba65264437eae/lib/elixir_analyzer/exercise_test/assert_call/compiler.ex#L57
(do to `[:Kernel, :SpecialForm]` what is done to `[:Kernel]`).

Hopefully that works, let me know if it doesn't :D

Reply:
I've changed the `modules_in_scope` value in the `assert/3` function, in compiler.ex to:
```elixir
modules_in_scope: %{
        [:Kernel] => Kernel.module_info(:exports),
        [:Kernel, :SpecialForms] => Kernel.SpecialForms.module_info(:exports)
      }
```

And the test suite to:
```elixir
assert_no_call "does not call the for/1 function" do
    type :essential
    called_fn module: Kernel.SpecialForms, name: :for
    comment Constants.dna_encoding_use_recursion()
  end
```

However, I still get two tests failing as `"elixir.dna-encoding.use_recursion"` isn't returned anymore :thinking:

Reply:
I looked into this more, and it seems I was wrong. `for` is special and cannot in fact be called with the fill prefix.

```elixir
iex(66)> for a <- [1,2,3], do: a
[1, 2, 3]
iex(67)> Kernel.SpecialForms.for a <- [1,2,3], do: a
** (CompileError) iex:67: undefined function <-/2
    (stdlib 3.15) lists.erl:1358: :lists.mapfoldl/3
```

The technical reason why this didn't work here is that `Kernel.SpecialForms` exposes `for` with arity 1, which didn't match the example (each `<-` and `do` add an argument). 

LEt's keep it the way you had it, it seems I have wasted your time on this one, I apologize.